test/test_binary_ufuncs.py

```diff
@@ -1359,100 +1359,109 @@ def test_rdiv(self, device, dtype):
+                    fns[1](x_, zero)
+            # Different value for different dtype on GPU
             else:
-                value = 255 if dtype == torch.uint8 else -1
-                self.assertTrue(torch.all(x.fmod(zero) == value))
-                # out
-                torch.fmod(x, zero, out=out)
-                self.assertTrue(torch.all(out == value))
-                self.assertEqual(out.size(), torch.Size([10, 10]))
-                # in-place
-                x_.fmod_(zero)
-                self.assertTrue(torch.all(x_ == value))
+                if dtype == torch.int64:
+                    self.assertEqual(fns[0](x, zero) == 4294967295, x >= 0)
+                    self.assertEqual(fns[0](x, zero) == -1, x < 0)
+                    # out
+                    fns[0](x, zero, out=out)
+                    self.assertEqual(out == 4294967295, x >= 0)
+                    self.assertEqual(out == -1, x < 0)
+                    self.assertEqual(out.size(), torch.Size([10, 10]))
+                    # in-place
+                    fns[1](x_, zero)
+                    self.assertEqual(x_ == 4294967295, x >= 0)
+                    self.assertEqual(x_ == -1, x < 0)
+                else:
+                    value = 255 if dtype == torch.uint8 else -1
+                    self.assertTrue(torch.all(fns[0](x, zero) == value))
+                    # out
+                    fns[0](x, zero, out=out)
+                    self.assertTrue(torch.all(out == value))
+                    self.assertEqual(out.size(), torch.Size([10, 10]))
+                    # in-place
+                    fns[1](x_, zero)
+                    self.assertTrue(torch.all(x_ == value))
 
     @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False, include_bool=False, include_complex=False))
-    def test_fmod(self, device, dtype):
+    def test_fmod_remainder(self, device, dtype):
         # Use numpy as reference
         def _reference_implementation(x, mod):
-            np_x = x.cpu().numpy()
-            np_mod = 0
-            if torch.is_tensor(mod):
-                np_mod = mod.cpu().numpy()
-            else:
-                np_mod = mod
-            exp = np.fmod(np_x, np_mod)
-            exp = torch.from_numpy(exp)
+            fn_list = [(torch.fmod, lambda x, mod: x.fmod_(mod), np.fmod),
+                       (torch.remainder, lambda x, mod: x.remainder_(mod), np.remainder)]
+            for fns in fn_list:
+                np_x = x.cpu().numpy()
+                np_mod = 0
+                if torch.is_tensor(mod):
+                    np_mod = mod.cpu().numpy()
+                else:
+                    np_mod = mod
+                exp = fns[2](np_x, np_mod)
+                exp = torch.from_numpy(exp)
 
-            res = torch.fmod(x, mod)
-            common_dtype = torch.promote_types(res.dtype, exp.dtype)
-            exp = exp.to(common_dtype)
+                res = fns[0](x, mod)
```

Comment:
Analogous comments about type handling here as in the fmod PR

Reply:
It would be more readable if the tuple was unpacked as `fn, inplace, ref` immediately instead of referencing its parts with indices.